db/db_impl/db_impl_write.cc

```diff
@@ -2114,48 +1936,11 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
 
   cfd->mem()->SetNextLogNumber(logfile_number_);
   assert(new_mem != nullptr);
-  // By default, it is assumed that the 'old' memtable
-  // will be added to the Imm memtable list and will therefore
-  // contribute to the Imm memory footprint.
-  bool noImmMemoryContribution = false;
-  // If MemPurge activated, purge and delete current memtable.
-  if (immutable_db_options_.experimental_allow_mempurge &&
-      ((cfd->GetFlushReason() == FlushReason::kOthers) ||
-       (cfd->GetFlushReason() == FlushReason::kManualFlush))) {
-    Status mempurge_s = MemPurge(cfd, new_mem);
-    if (mempurge_s.ok()) {
-      // If mempurge worked successfully,
-      // create sync point and decrement current memtable reference.
-      TEST_SYNC_POINT("DBImpl::MemPurge");
-      cfd->mem()->Unref();
-      // If the MemPurge is successful, the 'old' (purged) memtable
-      // is not added to the Imm memtable list and therefore
-      // does not contribute to the Imm memory cost anymore.
-      noImmMemoryContribution = true;
-    } else {
-      // If mempurge failed, go back to regular mem->imm->flush workflow.
-      assert(new_mem != nullptr);
-      delete new_mem;
-      SuperVersion* new_superversion =
-          context->superversion_context.new_superversion.release();
-      if (new_superversion != nullptr) {
-        delete new_superversion;
-      }
-      SequenceNumber seq = versions_->LastSequence();
-      new_mem = cfd->ConstructNewMemtable(mutable_cf_options, seq);
-      assert(new_mem != nullptr);
-      context->superversion_context.NewSuperVersion();
-      cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
-    }
-  } else {
-    // Else make the memtable immutable and proceed as usual.
-    cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
-  }
+  cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
```

Comment:
Remove changes made during #8454 when the mempurge process was happening in the middle of the SwitchMemtable process.